backend/src/user/model.ts

```diff
@@ -1,14 +1,20 @@
-import { Static, Record, String, Null, Unknown, Optional } from "runtypes";
+import { Static, Record, String, Null, Unknown, Optional, Literal } from "runtypes";
 import { Uuid, sanitizeUuid } from "./structures/Uuid";
 import { isObjectId } from "../db";
 
+export enum Role {
```

Comment:
Tutaj nie wiem czy nie lepiej zrobić to przy użyciu Literal tak jak np. Provider, wtedy ten constraint będzie sam z siebie działał. Widzę, że nawet coś próbowałeś bo został import to możesz napisać czemu jedank to zmieniłeś. I wyrzuciłbym to do structures.

Reply:
W opisie taska Marcin zaznaczył, żeby było to zrobione enumem, więc założyłem, że ma jakiś plan odnośnie tego. Literałem robiłem najpierw jak nie chciał zadziałać enum, żeby sprawdzić, czy w ogóle 'role' działa poprawnie

Reply:
Napisałem o enumie z myślą, że ten Literal jest też jakąś formą enuma (typu wyliczeniowego). Nie mam dużego doświadczenia z typescriptem, więc tu oddaje decyzję Mateuszowi jak to powinno wyglądać.

Reply:
Dobra zastanowiłem się nad tym i chyba faktycznie rozwiązanie z Union(Static) może nie być najlepsze w tym przypadku, bo trudno się dostać do pojedyńczych pól. Zamiast tego zaproponowałbym rozwiązanie hybrydowe, nieco podobne do tego pierwszego.

Całą implementację roli przenieść do nowego pliku w folderze `user/structures`. Plik ten eksportowałby dwie rzeczy:
1. enum tak jak jest teraz
2. Runtype z funkcją zawężającą typ do enuma

Wyglądałoby to mniej więcej tak:
```js
export enum RoleEnum { user = "USER", admin = "ADMIN" }

export const Role = String.withGuard((s): s is RoleEnum => {
	return Object.values(RoleEnum).includes(s as RoleEnum);
});
```

W tej sytuacji exportowanie typeof Role nie jest potrzebne bo każda poprawna rola będzie miała typ `RoleEnum`. Takie rozwiązanie będzie też odporne na ewentualne zmiany w enumie w przyszłości.

Reply:
Dobra, jutro to ogarnę, sorry za opóźnienie, ale się wbiłem w środek kolosów